gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/AbstractDiskRegion.java

```diff
@@ -775,7 +777,15 @@ public int getRecoveredEntryCount() {
   public void incRecoveredEntryCount() {
     this.recoveredEntryCount++;
   }
-  
+
+  public int getInvalidOrTombstoneEntryCount() {
+    return this.invalidOrTombstoneCount;
+  }
+
+  public void incInvalidOrTombstoneEntryCount() {
+    this.invalidOrTombstoneCount++;
+  }
+
```

Comment:
This is incremented when crf and krf is read at the startup. This is exactly on the lines of incRecoveredCount which is not threadsafe and is not required also. The compactor thread is already waiting when this oplog is read so no need of synchronization.